### PR TITLE
More tests for non-delta queries

### PR DESCRIPTION
**Pull request by meain**
This add a sanity tests with delta queries disabled as well as more unit and integration tests to check this behaviour.

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [x] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #<issue>

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: .github/workflows/sanity-test.yaml [hunk 1]
```diff
@@ -248,6 +248,53 @@ jobs:
           set -euo pipefail
           ./sanityTest
 
+      # non-delta backup
+      - name: Backup exchange incremental
```
Comment (ryanfkeepers) [inline .github/workflows/sanity-test.yaml:252]:
```suggestion
      - name: Backup exchange incremental w/out delta
```

File: .github/workflows/sanity-test.yaml [hunk 2]
```diff
@@ -248,6 +248,53 @@ jobs:
...
+        id: exchange-incremental-test
+        run: |
+          set -euo pipefail
+          echo -e "\nBackup Exchange incremental test\n" >> ${CORSO_LOG_FILE}
```
Comment (ryanfkeepers) [inline .github/workflows/sanity-test.yaml:256]:
```suggestion
          echo -e "\nBackup Exchange incremental test w/out delta\n" >> ${CORSO_LOG_FILE}
```

File: src/internal/connector/data_collections.go [hunk 3]
```diff
@@ -198,7 +200,7 @@ func checkServiceEnabled(
 		canMakeDeltaQueries = info.CanMakeDeltaQueries()
 	}
 
-	return true, canMakeDeltaQueries, nil
```
Comment (ryanfkeepers) [inline src/internal/connector/data_collections.go:201]: very minor: imo this intersection should occur when handling the result of checkServiceEnabled, instead of passing down the opts into this call.  It's not much difference, but it does keep concerns separate.  The concern of this func is to report on the results of looking up the resource owner info.  The concern of the caller is to evaluate the combination of passed in options and lookup results.

File: src/internal/connector/exchange/service_iterators_test.go [hunk 4]
```diff
@@ -1144,11 +1144,12 @@ func (suite *ServiceIteratorsSuite) TestFilterContainersAndFillCollections_incre
...
-		getter   mockGetter
-		resolver graph.ContainerResolver
-		dps      DeltaPaths
-		expect   map[string]endState
+		name            string
+		getter          mockGetter
+		resolver        graph.ContainerResolver
+		dps             DeltaPaths
+		expect          map[string]endState
+		skipForNonDelta bool
```
Comment (ryanfkeepers) [inline src/internal/connector/exchange/service_iterators_test.go:1152]: nit: the name here is slightly confusing
```suggestion
		skipWhenForcedNoDeltas bool
```

File: src/internal/connector/exchange/service_iterators_test.go [hunk 5]
```diff
@@ -1403,54 +1405,71 @@ func (suite *ServiceIteratorsSuite) TestFilterContainersAndFillCollections_incre
...
 
-			ctx, flush := tester.NewContext()
-			defer flush()
+			if canMakeDeltaQueries {
+				name += "-delta"
+			} else {
+				if test.skipForNonDelta {
+					continue
+				}
+				name += "-non-delta"
```
Comment (ryanfkeepers) [inline src/internal/connector/exchange/service_iterators_test.go:1421]: Instead of silently skipping, lets make this loud:
```suggestion
				name += "-non-delta"
				
				// move this just inside suite.Run, line 1426 or so.
				if test.skipForNonDelta {
					t.Skip("intentionally skipped non-delta case")
				}
```

**Review by ryanfkeepers — APPROVED** [review]
lgtm, just nits

File: src/internal/connector/data_collections.go (hunk 3 above)
Comment (meain) [reply src/internal/connector/data_collections.go:201]: This was moved out. Looks like I missed rebased with that commit in this PR.

File: src/internal/connector/exchange/service_iterators_test.go
Comment (ashmrtn) [inline src/internal/connector/exchange/service_iterators_test.go:1]: these tests help ensure coverage, but I think one more test (or test case) may still need to be added to cover an edge case. I think we need a test with
* DeltaPaths (previous backup metadata) equal to what non-deltas would produce, needs at least one folder
* container resolver returning all but one folder in the DeltaPath set
* ensuring the folder not returned was marked as deleted

From what I've gathered by skimming the test cases, these tests ensure mostly that running a non-delta backup works assuming the previous backup was a delta backup (i.e. input metadata is consistent with what would be generated by a delta backup). We'll need the other way around too since we want to rely on deltas where possible

Comment (meain) [reply src/internal/connector/exchange/service_iterators_test.go:1]: I've modified `TestFilterContainersAndFillCollections_incrementals_nondelta` to make sure that it test all combinations of before/after * delta/non-delta.
